### ISSUE TITLE
'reconfigure' required after update 7.11.4 => 7.12.0

### ISSUE THREAD

**Issue by bcambl** [gitlabhq/gitlabhq#9406]
I recently performed an update of gitlab-ce via yum on RHEL6.
(**gitlab-ce-7.11.4~omnibus-1.x86_64** to **gitlab-ce-7.12.0~omnibus-1.x86_64**)
This is the first time I have used yum to update gitlab since the older rpm update method.
Upon completing update via yum, there was a message stating that if gitlab does not function properly to try ```sudo gitlab-ctl restart```.

I was receiving the gitlab 500 page and found the sidekiq process consuming 100% CPU 

I tried the recommended  ```sudo gitlab-ctl restart``` with the same results.

After checking the log /var/log/gitlab/gitlab-rails/sidekiq.log I found the the service was starting then shutting down with the following error:
```
Sidetiq::Supervisor start
[Sidetiq] Sidetiq::Actor::Clock id: 47310580 initialize
[Sidetiq] Sidetiq::Actor::Handler id: 47287880 initialize
[Sidetiq] Sidetiq::Actor::Handler id: 47273940 initialize
....
 Permission denied @ rb_sysopen - /opt/gitlab/embedded/service/gitlab-rails/.gitlab_shell_secret
....
 INFO: [Sidetiq] Sidetiq::Actor::Clock id: 51659760 shutting down ...
 INFO: [Sidetiq] Sidetiq::Actor::Handler id: 51637240 shutting down ...
 INFO: [Sidetiq] Sidetiq::Actor::Handler id: 51619100 shutting down ...
```

Resolution:
```
sudo gitlab-ctl stop
sudo gitlab-ctl reconfigure
sudo gitlab-ctl start
```

-cheers

**Comment by stanhu** [comment]
What do you think, @jacobvosmaer?

/cc: @maxlazio

**Comment by maxlazio** [comment]
gitlab_shell_secret permissions didn't change since their creation so I don't suspect this being an issue with omnibus-gitlab.

**Comment by jacobvosmaer** [comment]
We did change something with gitlab_shell_secret, it was being bundled into the package accidentally before and we stopped doing that in 7.12.

https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests/356

I don't see how that would lead to this behavior though.

**Comment by maxlazio** [comment]
How is that a change related to the end user?

**Comment by jacobvosmaer** [comment]
@maxlazio I don't know but it touches the file.

I can reproduce the issue by 

- installing https://packages.gitlab.com/gitlab/gitlab-ce/packages/el/6/gitlab-ce-7.11.4~omnibus-1.x86_64.rpm 
- running `gitlab-ctl reconfigure`
- installing https://packages.gitlab.com/gitlab/gitlab-ce/packages/el/6/gitlab-ce-7.12.0~omnibus-1.x86_64.rpm
- running `gitlab-ctl tail sidekiq`

**Comment by maxlazio** [comment]
This same RHEL specific issue is [described in omnibus-gitlab issue tracker](https://gitlab.com/gitlab-org/omnibus-gitlab/issues/649) and the fix for it is released with 7.12.2.
